### PR TITLE
Hw7

### PR DESCRIPTION
**Pull request by duron81**
Hello! Please revise my HW7.
I additionally implemented lazy-loading for component Courses with feature-module.

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { ModalLoginComponent } from './shared/modal/modal-login/modal-login.component';
+import { EditCourseComponent } from './courses/edit-course/edit-course.component';
+
+
+const routes: Routes = [
+  { path: '', redirectTo: 'courses', pathMatch: 'full' },
+  { path: 'login', component: ModalLoginComponent},
+  { path: 'courses/new', component: AddCourseComponent, canActivate: [AuthGuard]},
+  { path: 'courses/:id', component: EditCourseComponent, canActivate: [AuthGuard] },
+  { path: 'courses', loadChildren: () => import('./feature/feature.module').then(m => m.FeatureModule) },
```
Comment (mol-knysh) [inline src/app/app-routing.module.ts:17]: It is better not to create separate objects for `courses/new` etc - you can try to use nested routes and children
And loading FeatureModule on the courses route - seems a little unlogical

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,8 +1,10 @@
 <app-header [userName]="userName">
 </app-header>
-<app-modal-login *ngIf="showLoginModal" (userLogged)="onUserLogged()"></app-modal-login>
-<app-courses *appHideIfNotAuthenticated>
+<!-- <app-breadcrumbs></app-breadcrumbs> -->
+<!-- <app-modal-login *ngIf="showLoginModal" (userLogged)="onUserLogged()"></app-modal-login> -->
+<!-- <app-courses *appHideIfNotAuthenticated>
```
Comment (mol-knysh) [inline src/app/app.component.html]: Please delete commented code if it not used

File: src/app/feature/feature.module.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import { FeatureRoutingModule } from './feature-routing.module';
+import { FeatureComponent } from './feature.component';
+
+
+@NgModule({
+  declarations: [
+    FeatureComponent
```
Comment (mol-knysh) [inline src/app/feature/feature.module.ts]: What does this component implement?

File: src/app/app.component.html (hunk 2 above)
Comment (duron81) [reply src/app/app.component.html]: Done

File: src/app/feature/feature.module.ts (hunk 3 above)
Comment (duron81) [reply src/app/feature/feature.module.ts]: Literally nothing, I used Feature Module for lazy-loading of the CoursesComponent. And that Feature Module use in its structure template of FeatureComponent, I have deleted that component

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (duron81) [reply src/app/app-routing.module.ts:17]: In my project structure `courses/new` is not a child of `courses`. All these components are rendered in the root component, that's why I didn't implement nested routes.
FeatureModule I used for lazy-loading of CoursesComponent, for me its unlogical just for practice purpose

**Pull request merged**